### PR TITLE
Auto-refresh states on log page

### PR DESCRIPTION
**Pull request by erikh**
@capoferro I could really use your input here.

This works. it's functionally complete, works great, etc.

But!

* it now needs two API calls per page load due to architecture ... issues. Please read the patch to see some of these things.
* it's not a very clean patch, i'm passing react components to others for example.

I could use some advice on how to improve this. Thanks!

### CODE REVIEW

File: src/logstats.js [hunk 1]
```diff
@@ -25,24 +25,60 @@ class CenterHeader extends React.Component {
...
+          var orig = this.state.run
+          this.setState({ run: result })
+          if (!orig) {
+            this.props.term.resize()
+          }
+        },
+        error => {
+          error
+            .then(res => res.errors[0])
+            .then(res => this.props.uiActions.processError(res))
```
Comment (capoferro) [inline src/logstats.js]: This is the same as `.then(this.props.uiActions.processError)`

File: src/logstats.js [hunk 2]
```diff
@@ -25,24 +25,60 @@ class CenterHeader extends React.Component {
...
+    this.timer = window.setInterval(this.refresh.bind(this), 5000)
+  }
+
+  componentWillUnmount() {
+    window.clearInterval(this.timer)
+  }
+
+  refresh() {
+    if (this.run_id) {
+      this.getAndUpdateState(
```
Comment (capoferro) [inline src/logstats.js]: Few thoughts here.

1. We're passing the log component in as `term` so we have access to `resize()`. That tightly couples these two components and makes them harder to maintain. Might as well make them one component if we're doing that so at least everything is in one spot.

2. We're making the `/uisvc/run/:run_id` in both log and logstats, but we're passing the run object from log to logstats anyway. Can we move `refresh()` to log?

If we do that, then both components will update when the refresh is called cause we pass the run as a prop to logstats anyway.

Further, that means we no longer need to pass the log component in as `term`, and can break that coupling.

3. We should add a `getRun()` to `/models/run` and use that to fetch the run.

Comment (erikh) [reply src/logstats.js]: ok, i'll do this later tonight.

File: src/logstats.js (hunk 1 above)
Comment (erikh) [reply src/logstats.js]: this was copied from edits  you made :)

**Comment by erikh** [comment]
PTAL; moved refresh to log.js, made refresh the control mech on which the rest of the log system works. props are now used for logstats.

File: src/logstats.js [hunk 3]
```diff
@@ -25,24 +25,25 @@ class CenterHeader extends React.Component {
 }
 
 class LogStats extends BaseComponent {
+  timer = null
+  run_id = null
+
   render() {
-    if (this.props.run === null) {
+    var run = this.props.run
```
Comment (capoferro) [inline src/logstats.js]: This should be `const` as it doesn't get reassigned.

File: src/log.js [hunk 4]
```diff
@@ -75,7 +87,7 @@ class Log extends BaseComponent {
       <Base
         repoName={this.state.run && this.state.run.task.ref.repository.name}
       >
-        <LogStats run={this.state.run} />
+        <LogStats term={this} run={this.state.run} />
```
Comment (capoferro) [inline src/log.js]: Do we still use `term` in `LogStats`? I don't see where we do, so perhaps this prop can be deleted?

Comment (erikh) [reply src/log.js]: yeah sorry overlooked it

**Comment by erikh** [comment]
PTAL @capoferro

**Pull request merged**